libs/milvus/langchain_milvus/vectorstores/milvus.py

```diff
@@ -856,6 +853,77 @@ def add_texts(
             logger.debug("Nothing to insert, skipping.")
             return []
 
+        return self.add_embeddings(
+            texts=texts,
+            embeddings=embeddings,
+            metadatas=metadatas,
+            timeout=timeout,
+            batch_size=batch_size,
+            ids=ids,
+            **kwargs,
+        )
+
+    def add_embeddings(
+        self,
+        texts: List[str],
+        embeddings: List[List[Union[float, List[float]]]],
+        metadatas: Optional[List[dict]] = None,
+        timeout: Optional[float] = None,
+        batch_size: int = 1000,
+        *,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Insert text data with embeddings vectors into Milvus.
+
+        Inserting data when the collection has not be made yet will result
+        in creating a new Collection. The data of the first entity decides
+        the schema of the new collection, the dim is extracted from the first
+        embedding and the columns are decided by the first metadata dict.
+        Metadata keys will need to be present for all inserted values. At
+        the moment there is no None equivalent in Milvus.
+
+        Args:
+            texts (List[str]): the texts to insert
+            embeddings (List[List[Union[float, List[float]]]]):
+                A vector embeddings for each text (in case of a single vector)
+                or list of vectors for each text (in case of multi-vector)
+            metadatas (Optional[List[dict]]): Metadata dicts attached to each of
+                the texts. Defaults to None.
+            should be less than 65535 bytes. Required and work when auto_id is False.
+            timeout (Optional[float]): Timeout for each batch insert. Defaults
+                to None.
+            batch_size (int, optional): Batch size to use for insertion.
+                Defaults to 1000.
+            ids (Optional[List[str]]): List of text ids. The length of each item
+
+        Raises:
+            MilvusException: Failure to add texts and embeddings
+
+        Returns:
+            List[str]: The resulting keys for each inserted element.
+        """
+        from pymilvus import Collection, MilvusException
+
+        transposed_embeddings = [
```

Comment:
not sure why transpose the embedding , and then transpose it back. it seems the same as the original embedding. if I’ve misunderstood anything just let me know

Reply:
@zc277584121 @MateuszOssGit  I tell you what is my opinion. Assume one wants to use **directly** the new `add_embeddings` method, and it has multi-vector indexes.

Lets assume we have [f1, f2] embeddings functions and ["a", "b", "c"] as texts, so the current (without the transpose) structure is:

```python
embeddings = [
    [f1(a), f1(b), f1(c)],
    [f2(a), f2(b), f2(c)]
]
```

So actually we require the user to pass the following arguments to `add_embeddings` method:
```python

embeddings = [
    [f1(a), f1(b), f1(c)],
    [f2(a), f2(b), f2(c)]
]
```

Which feels non-intuitive as opposed to the transposed version:
```
texts = ["a", "b", "c"]
transposed_embeddings = [
    [f1(a), f2(a)],
    [f1(b), f2(b)],
    [f1(c), f2(c)]
]
```
Where each text (i.e. "a") has its corresponding list of embeddings: `[f1(a), f2(a)]`.

If you agree, one way to overcome it and remove this transpose embeddings (which is indeed non-efficient) is to refactor the `add_embeddings` implementation to be text-oriented and not embedding-function oriented